compiler/backend/presLangScript.sml

```diff
@@ -234,381 +234,341 @@ val exh_to_pres_exp_def = tDefine"exh_to_pres_exp"`
   /\
-  (tctor_to_json TC_tup = String "TC_tup")
+  (tctor_to_structured TC_word8 = string_to_structured "TC_word8")
   /\
-  (tctor_to_json TC_exn = String "TC_exp")
+  (tctor_to_structured TC_word64 = string_to_structured "TC_word64")
   /\
-  (tctor_to_json TC_vector = String "TC_vector")
+  (tctor_to_structured TC_word8array = string_to_structured "TC_word8array")
   /\
-  (tctor_to_json TC_array = String "TC_array")`
-
-val t_to_json_def = tDefine"t_to_json"`
-  (t_to_json (Tvar tvarN) = String tvarN)
+  (tctor_to_structured TC_fn = string_to_structured "TC_fn")
   /\
-  (t_to_json (Tvar_db n) = num_to_json n)
+  (tctor_to_structured TC_tup = string_to_structured "TC_tup")
   /\
-  (t_to_json (Tapp ts tctor) = Object [("Tapp", Array (MAP t_to_json ts));
-  ("tctor", tctor_to_json tctor)])`
-  cheat;
-
-val num_to_hex_digit_def = Define `
-  num_to_hex_digit n =
-    if n < 10 then [CHR (48 + n)] else
-    if n < 16 then [CHR (55 + n)] else []`;
-
-val num_to_hex_def = Define `
-  num_to_hex n =
-    (if n < 16 then [] else num_to_hex (n DIV 16)) ++
-    num_to_hex_digit (n MOD 16)`;
-
-val word_to_hex_string_def = Define `
-  word_to_hex_string w = "0x" ++ num_to_hex (w2n (w:'a word))`;
-
-val lit_to_json_def = Define`
-  (lit_to_json (IntLit i) = new_obj "IntLit" [("value", Int i)])
+  (tctor_to_structured TC_exn = string_to_structured "TC_exp")
   /\
-  (lit_to_json (Char c) = new_obj "Char" [("value", String [c])])
+  (tctor_to_structured TC_vector = string_to_structured "TC_vector")
   /\
-  (lit_to_json (StrLit s) = new_obj "StrLit" [("value", String s)])
+  (tctor_to_structured TC_array = string_to_structured "TC_array")`
+
+val t_to_structured_def = tDefine"t_to_json"`
+  (t_to_structured (Tvar tvarN) = Item NONE "Tvar" [string_to_structured tvarN])
   /\
-  (lit_to_json (Word8 w) = new_obj "Word8" [("value", String (word_to_hex_string w))])
+  (t_to_structured (Tvar_db n) = Item NONE "Tvar_db" [num_to_structured n])
   /\
-  (lit_to_json (Word64 w) = new_obj "Word64" [("value",  String (word_to_hex_string w))])`
+  (t_to_structured (Tapp ts tctor) = Item NONE "Tapp" [ List (MAP t_to_structured ts);
+    tctor_to_structured tctor])`
+  cheat;
 
-val option_to_json_def = Define`
-  (option_to_json opt = case opt of
-                      | NONE => Null
-                      | SOME opt' => String opt')`
+val tid_or_exn_to_structured_def = Define`
+  tid_or_exn_to_structured te =
+   let (name, id) =
+     case te of
+       | TypeId id =>  ("TypeId", id)
+       | TypeExn id => ("TypeExn", id) in
+     Item NONE name [id_to_structured id]`;
+
+val conf_to_structured_def = Define`
+  conf_to_structured con =
+    let none = Item NONE "NONE" [] in
+      case con of
+         | Modlang_con NONE => none
+         | Conlang_con NONE => none
+         | Modlang_con (SOME id) => Item NONE "SOME" [id_to_structured id]
+         | Conlang_con (SOME (n,t)) => Item NONE "SOME" [Tuple [num_to_structured
+         n; tid_or_exn_to_structured t]]
```

Comment:
Nit: don't break the line at this point, i.e. join this line and the one above it.